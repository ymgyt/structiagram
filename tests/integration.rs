use assert_cmd::Command;

#[test]
fn case1() {
    let mut bin = bin();

    let mut path = fixtures();
    path.push("case1");
    bin.args(["--dir", path.display().to_string().as_str()]);

    insta::assert_snapshot!(&String::from_utf8_lossy(
        bin.assert().success().get_output().stdout.as_slice(),
    ));
}

// asser_cmd create temporary directory for testing command
// so we need absolute path.
fn fixtures() -> std::path::PathBuf {
    std::fs::canonicalize("./tests/fixtures").unwrap()
}
fn bin() -> Command {
    Command::cargo_bin("structiagram").unwrap()
}
