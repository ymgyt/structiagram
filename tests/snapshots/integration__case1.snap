---
source: tests/integration.rs
expression: "&String::from_utf8_lossy(bin.assert().success().get_output().stdout.as_slice())"
---
# Structiagram

```mermaid
%%{init: {
            "er": {
                "layoutDirection": "LR",
                "entityPadding": 15,
                "useMaxWidth": false
            }
        }}%%
erDiagram
A {
    String field_1
    BId b_id
}
B {
    Option_String field_1
}
A ||--|| B : b_id
```

